python/GafferSceneUITest/LightEditorTest.py

```diff
@@ -656,6 +660,51 @@ def testLightMuteAttribute( toggleCount, toggleLocation, newStates ) :
 
 			del widget, editor
 
+	def testToggleContext( self ) :
+
+		# Make sure the correct context is scoped when evaluating attributes
+		# to determine a new toggle value
+
+		script = Gaffer.ScriptNode()
+		script["variables"].addChild( Gaffer.NameValuePlug( "test", IECore.FloatData( 5.0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic ) )
+
+		script["light"] = GafferSceneTest.TestLight()
+
+		script["group"] = GafferScene.Group()
+		script["group"]["in"][0].setInput( script["light"]["out"] )
+
+		script["filter"] = GafferScene.PathFilter()
+		script["filter"]["paths"].setValue( IECore.StringVectorData( ["/group"] ) )
+
+		script["custAttr"] = GafferScene.CustomAttributes()
+		script["custAttr"]["in"].setInput( script["group"]["out"] )
+		script["custAttr"]["filter"].setInput( script["filter"]["out"] )
+		script["custAttr"]["attributes"].addChild( Gaffer.NameValuePlug( "gl:visualiser:scale", IECore.FloatData( 2.0 ), "scale", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic ) )
+
+		script["expression"] = Gaffer.Expression()
+		script["expression"].setExpression(
+			'parent["custAttr"]["attributes"]["scale"]["value"] = context["test"]',
+			"python"
+		)
+
+		self.assertEqual( script.context().get( "test" ), 5.0 )
+
+		with script.context() :
+			attr = script["custAttr"]["out"].attributes( "/group" )
+
+		self.assertIn( "gl:visualiser:scale", attr )
+		self.assertEqual( attr["gl:visualiser:scale"].value, 5.0 )
+
+		editor = GafferSceneUI.LightEditor( script )
+		widget = editor._LightEditor__pathListing
+		editor._LightEditor__settingsNode["in"].setInput( script["custAttr"]["out"] )
```

Comment:
With this removed, the test will pass _both_ before and after the proposed fix.

With it included, it only passes with the fix.

I expect that without this line, the test would fail both before the fix (and maybe after too?). Where is it getting the context variable `test`?

Reply:
> Where is it getting the context variable test?

I don't think it is getting the context variable. Without the `setInput()` call, it's not showing anything at all, so it probably never calls `fullAttributes()`. `_LightEditor__settingsNode` is a bit of a hacky way of doing this though - we should just use the public API and call `editor.setNodeSet( Gaffer.StandardSet( [ script["custAttr"] ) )`.

Reply:
That makes sense. I changed to using `editor.setNodeSet()` and squashed into https://github.com/GafferHQ/gaffer/pull/5408/commits/d647680ee95b87544412fe504968d68e9d10f69e